app/views/distributions/_form.html.erb

```diff
@@ -1,3 +1,11 @@
+<%
+  partner_list = if defined?(source) && source == 'new'
```

Comment:
This actually shouldn't be calculated by the view at all. It's better to pass this in from whatever view is calling the partial.

Very simple DB queries are _kind of_ OK within views, but in general we should be aiming not to have views talk to the DB at all.

Reply:
that makes sense. I'll move the query into the controller actions and pass them from there.